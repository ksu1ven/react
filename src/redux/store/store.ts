import { configureStore } from '@reduxjs/toolkit';
import paginationSlice from '../features/paginationSlice';
import limitSlice from '../features/limitSlice';
import searchSlice from '../features/searchSlice';
import loaderSlice from '../features/loaderSlice';
import resultsSlice from '../features/resultsSlice';
import { cardsApi } from '../api/searchCards';

export const store = configureStore({
  reducer: {
    pagination: paginationSlice,
    limit: limitSlice,
    search: searchSlice,
    loader: loaderSlice,
    results: resultsSlice,
    [cardsApi.reducerPath]: cardsApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(cardsApi.middleware),
});

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;
