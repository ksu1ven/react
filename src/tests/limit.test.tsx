import { describe, expect, it, vi } from 'vitest';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../redux/store/store';
import SelectLimit from '../components/SearchPage/Select';

describe('Tests for the Select component', () => {
  it('Value changes correctly', async () => {
    const limit = render(
      <Provider store={store}>
        <SelectLimit params={new URLSearchParams()} setParams={vi.fn()} />
      </Provider>
    );
    const select = screen.getByTestId('select');
    expect(select).toHaveValue('10');
    fireEvent.change(select, { target: { value: '5' } });
    expect(select).toHaveValue('5');
    expect(limit.baseElement).toMatchSnapshot();
  });
});
