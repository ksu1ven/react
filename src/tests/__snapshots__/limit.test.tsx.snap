// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for the Select component > Value changes correctly 1`] = `
<body>
  <div>
    <label
      class="flex justify-end mr-32 font-extrabold text-xl"
    >
      Limit per page:
      <select
        class="bg-lime-700 text-white"
        data-testid="select"
        id="itemsPerPage"
        name="itemsPerPage"
      >
        <option
          data-testid="select-option"
          value="5"
        >
          5
        </option>
        <option
          data-testid="select-option"
          value="10"
        >
          10
        </option>
        <option
          data-testid="select-option"
          value="20"
        >
          20
        </option>
        <option
          data-testid="select-option"
          value="50"
        >
          50
        </option>
        <option
          data-testid="select-option"
          value="100"
        >
          100
        </option>
      </select>
    </label>
  </div>
</body>
`;
